yubikey-luks-enroll

```diff
@@ -62,50 +62,91 @@ echo "This script will utilize the YubiKey slot $YUBIKEY_LUKS_SLOT for slot $SLO
 
 if [ "$CLEAR_SLOT" = "1" ]; then
     echo "Killing LUKS slot $SLOT"
-    cryptsetup luksKillSlot "$DISK" "$SLOT"
+    yubikey-luks-remove luksKillSlot "$DISK" "$SLOT"
 fi
 
-echo "Adding yubikey to initrd"
-
-while true ; do
-    if lsusb | grep -iq 'yubico'; then break; fi
-    printf "Please insert a yubikey and press enter."
-    read -r _ <&1
-done
-
-P1=$(/lib/cryptsetup/askpass "Please enter the yubikey challenge password. This is the password that will only work while your yubikey is installed in your computer:")
+P1=$(/lib/cryptsetup/askpass "Please enter the new passphrase or yubikey challenge password. If using a Yubikey, this is the password that will only work while your yubikey is installed in your computer:")
 if [ "$DBG" = "1" ]; then echo "Password: $P1"; fi
 
-P2=$(/lib/cryptsetup/askpass "Please enter the yubikey challenge password again:")
+P2=$(/lib/cryptsetup/askpass "Please enter the new passphrase or yubikey challenge password again:")
 if [ "$DBG" = "1" ]; then echo "Password: $P2"; fi
 
 if [ "$P1" != "$P2" ]; then
     echo "Passwords do not match"
     exit 1
 fi
 
-if [ "$HASH" = "1" ]; then
-    P1=$(printf %s "$P1" | sha256sum | awk '{print $1}')
-    if [ "$DBG" = "1" ]; then echo "Password hash: $P1"; fi
-fi
+read -p "Are you using a Yubikey with this passphrase? [Y/n]  " USE_YUBIKEY1
 
-R="$(printf %s "$P1" | ykchalresp -"$YUBIKEY_LUKS_SLOT" -i- 2>/dev/null || true)"
-if [ "$DBG" = "1" ]; then echo "Yubikey response: $R"; fi
+if [ "$USE_YUBIKEY1" = "y" ] || [ "$USE_YUBIKEY1" = "Y" ]; then
+    while true ; do
+        if lsusb | grep -iq 'yubico'; then break; fi
```

Comment:
This still doesn't guarantee the chose slot is properly configured. Take a look how it's checked [here](https://github.com/agherzan/yubikey-full-disk-encryption/blob/master/src/ykfde-enroll#L87).